azure-container-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/containerapps/DeployMojo.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.maven.containerapps;
+
+import com.fasterxml.jackson.dataformat.javaprop.JavaPropsMapper;
+import com.fasterxml.jackson.dataformat.javaprop.JavaPropsSchema;
+import com.microsoft.azure.maven.utils.MavenConfigUtils;
+import com.microsoft.azure.toolkit.lib.common.exception.AzureToolkitRuntimeException;
+import com.microsoft.azure.toolkit.lib.common.operation.AzureOperation;
+import com.microsoft.azure.toolkit.lib.containerapps.config.ContainerAppConfig;
+import com.microsoft.azure.toolkit.lib.containerapps.task.DeployContainerAppTask;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+
+import java.io.IOException;
+
+import static com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES;
+
+/**
+ * Deploy your project to target Azure Container app. If target app doesn't exist, it will be created.
+ */
+@Mojo(name = "deploy")
+@Slf4j
+public class DeployMojo extends AbstractMojoBase {
+    /**
+     * Boolean flag to control whether to wait the deployment status to be ready after deployment
+     */
+    @Parameter(property = "noWait")
+    private Boolean noWait;
```

Comment:
Did we use the two parameters anywhere else? Shall we remove them?

Reply:
Removed